deviceplugin/deviceplugin.go

```diff
@@ -0,0 +1,144 @@
+// Package deviceplugin provides a basic implementation of a Kubernetes DevicePlugin without content.
+// It is intended to be used to build other DevicePlugins.
+package deviceplugin
+
+import (
+	"log"
+	"net"
+	"os"
+	//      "strings"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	"path"
+	"time"
+)
+
+type DevicePlugin struct {
```

Comment:
It is better to expose an interface and create a struct to implement.

e.g.

```go
type DevicePlugin interface {
    Foo() uint64
}

type devicePlugin struct {}

func (plugin *devicePlugin) Foo() uint64 {
    return 0
}

func NewDevicePlugin() DevicePlugin {
    return &devicePlugin{}
}
```
This allows us to easily mock the interface for testing and allows our code to be generic and work with other implementations that implement the interface.

Reply:
So if I were actually doing an interface, definitely.  But I am also trying to create the moral equivalent of an abstract base, where a lot of the repetitive work is abstracted out of being the concern of the author of various Device Plugins.  I.e.  You will need to handle your own logic around Allocate et al, but Start(), Stop() etc can be done in a single standard way.  That was the thought anyway, I'm open to other approaches :)